--- php-7.2.0/configure.ac.fixheader	2017-04-12 09:42:21.510812285 +0200
+++ php-7.2.0/configure.ac	2017-04-12 09:43:45.055224385 +0200
@@ -1263,10 +1263,10 @@
 EXTRA_LDFLAGS="$EXTRA_LDFLAGS $PHP_LDFLAGS"
 EXTRA_LDFLAGS_PROGRAM="$EXTRA_LDFLAGS_PROGRAM $PHP_LDFLAGS"
 
-PHP_BUILD_DATE=`date '+%Y-%m-%d'`
+PHP_BUILD_DATE=`date '+%Y-%m-%d' -r $srcdir/NEWS`
 AC_DEFINE_UNQUOTED(PHP_BUILD_DATE,"$PHP_BUILD_DATE",[PHP build date])
 
-PHP_UNAME=`uname -a | xargs`
+PHP_UNAME=`uname | xargs`
 AC_DEFINE_UNQUOTED(PHP_UNAME,"$PHP_UNAME",[uname -a output])
 PHP_OS=`uname | xargs`
 AC_DEFINE_UNQUOTED(PHP_OS,"$PHP_OS",[uname output])
