
https://bugzilla.redhat.com/bugzilla/attachment.cgi?id=135828

--- php-4.3.9/ext/standard/metaphone.c.metaphone
+++ php-4.3.9/ext/standard/metaphone.c
@@ -152,7 +152,7 @@
 						(*phoned_word)[p_idx++] = c; \
 					}
 /* Slap a null character on the end of the phoned word */
-#define End_Phoned_Word	{(*phoned_word)[p_idx] = '\0';}
+#define End_Phoned_Word	Phonize('\0')
 /* How long is the phoned word? */
 #define Phone_Len	(p_idx)
 
