diff -up php-5.3.0/ext/openssl/openssl.c.openssl php-5.3.0/ext/openssl/openssl.c
--- php-5.3.0/ext/openssl/openssl.c.openssl	2009-04-20 11:44:29.000000000 +0200
+++ php-5.3.0/ext/openssl/openssl.c	2009-08-25 18:16:30.000000000 +0200
@@ -502,8 +502,13 @@ inline static int php_openssl_safe_mode_
 static char default_ssl_conf_filename[MAXPATHLEN];
 
 struct php_x509_request { /* {{{ */
+#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+	LHASH_OF(CONF_VALUE) * global_config;	/* Global SSL config */
+	LHASH_OF(CONF_VALUE) * req_config;	/* SSL config for this request */
+#else
 	LHASH * global_config;	/* Global SSL config */
 	LHASH * req_config;		/* SSL config for this request */
+#endif
 	const EVP_MD * md_alg;
 	const EVP_MD * digest;
 	char	* section_name,
@@ -680,7 +685,11 @@ static time_t asn1_time_to_time_t(ASN1_U
 }
 /* }}} */
 
+#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+static inline int php_openssl_config_check_syntax(const char * section_label, const char * config_filename,	const char * section, LHASH_OF(CONF_VALUE) * config TSRMLS_DC) /* {{{ */
+#else
 static inline int php_openssl_config_check_syntax(const char * section_label, const char * config_filename,	const char * section, LHASH * config TSRMLS_DC) /* {{{ */
+#endif
 {
 	X509V3_CTX ctx;
 	
