
Make sure odbc.so is linked against -lodbc etc.

--- php-4.3.1/ext/odbc/config.m4.odbc	2002-11-12 17:31:39.000000000 +0000
+++ php-4.3.1/ext/odbc/config.m4	2003-05-16 09:54:22.000000000 +0100
@@ -429,6 +429,8 @@
     ODBC_TYPE=unixODBC
     AC_DEFINE(HAVE_UNIXODBC,1,[ ])
     AC_MSG_RESULT(yes)
+    ODBC_SHARED_LIBADD="-lodbc -lodbcpsql"
+    PHP_SUBST(ODBC_SHARED_LIBADD)
   else
     AC_MSG_RESULT(no)
   fi
