From 78247eb2209b14c3c1273189dfa2615e2df21214 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Mon, 7 Oct 2019 16:31:09 +0200
Subject: [PATCH] add librt for opcache

---
 ext/opcache/config.m4 | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/ext/opcache/config.m4 b/ext/opcache/config.m4
index a388dd787b67..931e27b010f5 100644
--- a/ext/opcache/config.m4
+++ b/ext/opcache/config.m4
@@ -13,6 +13,9 @@ PHP_ARG_ENABLE([huge-code-pages],
 
 if test "$PHP_OPCACHE" != "no"; then
 
+  dnl Always build as shared extension
+  ext_shared=yes
+
   if test "$PHP_HUGE_CODE_PAGES" = "yes"; then
     AC_DEFINE(HAVE_HUGE_CODE_PAGES, 1, [Define to enable copying PHP CODE pages into HUGE PAGES (experimental)])
   fi
@@ -248,6 +251,9 @@ int main() {
 	Optimizer/zend_dump.c,
 	shared,,-DZEND_ENABLE_STATIC_TSRMLS_CACHE=1,,yes)
 
+  PHP_CHECK_LIBRARY(rt, shm_unlink, [PHP_ADD_LIBRARY(rt,1,OPCACHE_SHARED_LIBADD)])
+
   PHP_ADD_BUILD_DIR([$ext_builddir/Optimizer], 1)
   PHP_ADD_EXTENSION_DEP(opcache, pcre)
+  PHP_SUBST(OPCACHE_SHARED_LIBADD)
 fi
