diff -up php-7.2.3RC1/configure.ac.fixheader php-7.2.3RC1/configure.ac
--- php-7.2.3RC1/configure.ac.fixheader	2018-02-14 06:09:37.792429866 +0100
+++ php-7.2.3RC1/configure.ac	2018-02-14 06:10:33.755701362 +0100
@@ -1272,7 +1272,7 @@ EXTRA_LDFLAGS_PROGRAM="$EXTRA_LDFLAGS_PR
 PHP_BUILD_DATE=`date --utc --date="@${SOURCE_DATE_EPOCH:-$(date +%s)}" +%Y-%m-%d`
 AC_DEFINE_UNQUOTED(PHP_BUILD_DATE,"$PHP_BUILD_DATE",[PHP build date])
 
-PHP_UNAME=`uname -a | xargs`
+PHP_UNAME=`uname | xargs`
 AC_DEFINE_UNQUOTED(PHP_UNAME,"$PHP_UNAME",[uname -a output])
 PHP_OS=`uname | xargs`
 AC_DEFINE_UNQUOTED(PHP_OS,"$PHP_OS",[uname output])
