diff -up php5.3-200908150630/configure.in.gnusrc php5.3-200908150630/configure.in
--- php5.3-200908150630/configure.in.gnusrc	2009-08-12 00:34:39.000000000 +0200
+++ php5.3-200908150630/configure.in	2009-08-15 08:46:09.000000000 +0200
@@ -58,6 +58,8 @@ AC_DEFUN([PHP_EXT_DIR],[ext/$1])dnl
 AC_DEFUN([PHP_EXT_SRCDIR],[$abs_srcdir/ext/$1])dnl
 AC_DEFUN([PHP_ALWAYS_SHARED],[])dnl
 
+AC_DEFINE([_GNU_SOURCE], 1, [Define to enable GNU C Library extensions])
+
 dnl Setting up the PHP version based on the information above.
 dnl -------------------------------------------------------------------------
 
diff -up php5.3-200908150630/ext/interbase/interbase.c.gnusrc php5.3-200908150630/ext/interbase/interbase.c
--- php5.3-200908150630/ext/interbase/interbase.c.gnusrc	2009-07-17 14:23:09.000000000 +0200
+++ php5.3-200908150630/ext/interbase/interbase.c	2009-08-15 08:46:09.000000000 +0200
@@ -24,7 +24,6 @@
 #include "config.h"
 #endif
 
-#define _GNU_SOURCE
 
 #include "php.h"
 
diff -up php5.3-200908150630/ext/pdo_firebird/firebird_driver.c.gnusrc php5.3-200908150630/ext/pdo_firebird/firebird_driver.c
--- php5.3-200908150630/ext/pdo_firebird/firebird_driver.c.gnusrc	2009-07-17 14:22:08.000000000 +0200
+++ php5.3-200908150630/ext/pdo_firebird/firebird_driver.c	2009-08-15 08:46:09.000000000 +0200
@@ -22,7 +22,6 @@
 #include "config.h"
 #endif
 
-#define _GNU_SOURCE
 
 #include "php.h"
 #ifdef ZEND_ENGINE_2
diff -up php5.3-200908150630/ext/standard/file.c.gnusrc php5.3-200908150630/ext/standard/file.c
--- php5.3-200908150630/ext/standard/file.c.gnusrc	2009-07-17 14:20:59.000000000 +0200
+++ php5.3-200908150630/ext/standard/file.c	2009-08-15 08:46:09.000000000 +0200
@@ -123,9 +123,6 @@ php_file_globals file_globals;
 #endif
 
 #if defined(HAVE_FNMATCH) && !defined(PHP_WIN32)
-# ifndef _GNU_SOURCE
-#  define _GNU_SOURCE
-# endif
 # include <fnmatch.h>
 #endif
 
diff -up php5.3-200908150630/ext/zlib/zlib_fopen_wrapper.c.gnusrc php5.3-200908150630/ext/zlib/zlib_fopen_wrapper.c
--- php5.3-200908150630/ext/zlib/zlib_fopen_wrapper.c.gnusrc	2009-07-17 14:20:18.000000000 +0200
+++ php5.3-200908150630/ext/zlib/zlib_fopen_wrapper.c	2009-08-15 08:47:43.000000000 +0200
@@ -19,8 +19,6 @@
 
 /* $Id: zlib_fopen_wrapper.c 274034 2009-01-20 15:41:23Z felipe $ */
 
-#define _GNU_SOURCE
-
 #include "php.h"
 #include "php_zlib.h"
 #include "fopen_wrappers.h"
diff -up php5.3-200908150630/main/php.h.gnusrc php5.3-200908150630/main/php.h
--- php5.3-200908150630/main/php.h.gnusrc	2009-07-17 14:23:41.000000000 +0200
+++ php5.3-200908150630/main/php.h	2009-08-15 08:46:09.000000000 +0200
@@ -30,6 +30,7 @@
 #define PHP_HAVE_STREAMS
 #define YYDEBUG 0
 
+#include "php_config.h"
 #include "php_version.h"
 #include "zend.h"
 #include "zend_qsort.h"
diff -up php5.3-200908150630/main/streams/cast.c.gnusrc php5.3-200908150630/main/streams/cast.c
--- php5.3-200908150630/main/streams/cast.c.gnusrc	2009-07-17 14:23:41.000000000 +0200
+++ php5.3-200908150630/main/streams/cast.c	2009-08-15 08:48:09.000000000 +0200
@@ -18,7 +18,6 @@
 
 /* $Id: cast.c 279036 2009-04-20 08:28:44Z pajoye $ */
 
-#define _GNU_SOURCE
 #include "php.h"
 #include "php_globals.h"
 #include "php_network.h"
diff -up php5.3-200908150630/main/streams/streams.c.gnusrc php5.3-200908150630/main/streams/streams.c
--- php5.3-200908150630/main/streams/streams.c.gnusrc	2009-07-17 14:23:40.000000000 +0200
+++ php5.3-200908150630/main/streams/streams.c	2009-08-15 08:49:04.000000000 +0200
@@ -21,7 +21,6 @@
 
 /* $Id: streams.c 280678 2009-05-17 14:58:10Z lbarnaud $ */
 
-#define _GNU_SOURCE
 #include "php.h"
 #include "php_globals.h"
 #include "php_network.h"
diff -up php5.3-200908150630/Zend/zend_language_parser.c.gnusrc php5.3-200908150630/Zend/zend_language_parser.c
--- php5.3-200908150630/Zend/zend_language_parser.c.gnusrc	2009-08-15 08:36:38.000000000 +0200
+++ php5.3-200908150630/Zend/zend_language_parser.c	2009-08-15 08:46:09.000000000 +0200
@@ -366,6 +366,8 @@
 #include "zend_API.h"
 #include "zend_constants.h"
 
+#include <string.h>
+
 
 #define YYERROR_VERBOSE
 #define YYSTYPE znode
--- php-5.3.1RC2/main/streams/memory.c.gnusrc	2009-10-09 19:45:03.000000000 +0200
+++ php-5.3.1RC2/main/streams/memory.c	2009-10-21 17:31:50.000000000 +0200
@@ -18,7 +18,6 @@
 
 /* $Id: memory.c 289437 2009-10-09 17:45:03Z pajoye $ */
 
-#define _GNU_SOURCE
 #include "php.h"
 
 PHPAPI int php_url_decode(char *str, int len);
