diff -up php-5.2.3/sapi/embed/config.m4.embed php-5.2.3/sapi/embed/config.m4
--- php-5.2.3/sapi/embed/config.m4.embed	2005-05-07 04:51:53.000000000 +0200
+++ php-5.2.3/sapi/embed/config.m4	2007-08-10 13:46:30.000000000 +0200
@@ -11,7 +11,7 @@ AC_ARG_ENABLE(embed,
   case $enableval in
     yes|shared)
       PHP_EMBED_TYPE=shared
-      INSTALL_IT="\$(mkinstalldirs) \$(INSTALL_ROOT)\$(prefix)/lib; \$(INSTALL) -m 0755 $SAPI_SHARED \$(INSTALL_ROOT)\$(prefix)/lib"
+      INSTALL_IT="\$(mkinstalldirs) \$(INSTALL_ROOT)\$(libdir); \$(INSTALL) -m 0755 libs/libphp5-$PHP_VERSION.$SHLIB_DL_SUFFIX_NAME \$(INSTALL_ROOT)\$(libdir); \$(LN_S) libphp5-$PHP_VERSION.$SHLIB_DL_SUFFIX_NAME \$(INSTALL_ROOT)\$(libdir)/libphp5.$SHLIB_DL_SUFFIX_NAME"
       ;;
     static)
       PHP_EMBED_TYPE=static
diff -up php-5.2.3/Makefile.global.embed php-5.2.3/Makefile.global
--- php-5.2.3/Makefile.global.embed	2007-08-10 13:19:04.000000000 +0200
+++ php-5.2.3/Makefile.global	2007-08-10 13:27:27.000000000 +0200
@@ -14,7 +14,7 @@ all: $(all_targets) 
 build-modules: $(PHP_MODULES)
 
 libphp5.la: $(PHP_GLOBAL_OBJS) $(PHP_SAPI_OBJS)
-	$(LIBTOOL) --mode=link $(CC) $(CFLAGS) $(EXTRA_CFLAGS) -rpath $(phptempdir) $(EXTRA_LDFLAGS) $(LDFLAGS) $(PHP_RPATHS) $(PHP_GLOBAL_OBJS) $(PHP_SAPI_OBJS) $(EXTRA_LIBS) $(ZEND_EXTRA_LIBS) -o $@
+	$(LIBTOOL) --mode=link $(CC) $(CFLAGS) $(EXTRA_CFLAGS) -rpath $(phptempdir) -release $(PHP_VERSION) $(EXTRA_LDFLAGS) $(LDFLAGS) $(PHP_RPATHS) $(PHP_GLOBAL_OBJS) $(PHP_SAPI_OBJS) $(EXTRA_LIBS) $(ZEND_EXTRA_LIBS) -o $@
 	-@$(LIBTOOL) --silent --mode=install cp libphp5.la $(phptempdir)/libphp5.la >/dev/null 2>&1
 
 libs/libphp5.bundle: $(PHP_GLOBAL_OBJS) $(PHP_SAPI_OBJS)
@@ -25,13 +25,9 @@ install: $(all_targets) $(install_target
 install-sapi: $(OVERALL_TARGET)
 	@echo "Installing PHP SAPI module:       $(PHP_SAPI)"
 	-@$(mkinstalldirs) $(INSTALL_ROOT)$(bindir)
-	-@if test ! -r $(phptempdir)/libphp5.$(SHLIB_DL_SUFFIX_NAME); then \
-		for i in 0.0.0 0.0 0; do \
-			if test -r $(phptempdir)/libphp5.$(SHLIB_DL_SUFFIX_NAME).$$i; then \
-				$(LN_S) $(phptempdir)/libphp5.$(SHLIB_DL_SUFFIX_NAME).$$i $(phptempdir)/libphp5.$(SHLIB_DL_SUFFIX_NAME); \
-				break; \
-			fi; \
-		done; \
+	-@if test ! -r $(phptempdir)/libphp5.$(SHLIB_DL_SUFFIX_NAME) -a \
+		-r $(phptempdir)/libphp5-$(PHP_VERSION).$(SHLIB_DL_SUFFIX_NAME); then \
+		$(LN_S) $(phptempdir)/libphp5-$(PHP_VERSION).$(SHLIB_DL_SUFFIX_NAME) $(phptempdir)/libphp5.$(SHLIB_DL_SUFFIX_NAME); \
 	fi
 	@$(INSTALL_IT)
 
